### DISCUSSION TITLE
All tests ok in cypress.open, but when using cypress.run if 1 test fails, all subsequent tests fail

### DISCUSSION THREAD

**Discussion by andrewwhiteart** [cypress-io/cypress#28124] [Questions and Help]
i have testIsolation: false. cypress 13.3
I am using salesforce, i login fine in my before hook then i navigate to the page under test.
Then i have tests for each field on the page, i have left two in the example provided but there are about 30 tests.
If i run this and all 30 tests pass - all is well in both cypress run and open.
But if any test fails, cypress open handles this (only that test fails)
Where as with cypress run once a failure is encountered EVERY subsequent test fails.

It appears as though each subsequent test after a failed test no longer has access to the DOM, the error provided is this:
TypeError: Timed out retrying after 4000ms: Cannot read properties of undefined (reading 'length')
and if i change one of the subsequent tests to simply locate something on the page which is always present, it cannot find it. 

I am totally stuck, any help is much appreciated
 

```context('Check relevant content is present > in Salesforce', function() {

    before("Login", function() {
        Cypress.session.clearAllSavedSessions()
        cy.loginIntoSF("underwriter")
        cy.search_for_id_and_select_case_form('276345725')
    })

    describe('Check Salesforce > Case form > Other Applicant', function() {

        it("Details > applicant > Account Name", function () {
            cy.get(`[field-label="Account Name"]`)
                .should('be.visible')
                .contains('Ivor Dewdney')
        })

        it("Details > applicant > App Member", function () {
            cy.get(`[field-label="App Member"]`)
                .should('be.visible')
                .contains('Ivor Dewdney')
        })
    })
})

**Comment by portyu9** [comment]
Hi @andrewwhiteart 

The main problem is that the tests are coupled together through shared browser/application state.

You have:

```js
testIsolation: false
```

and the login/navigation happens only once in:

```js
before(...)
```

That means all ~30 `it()` blocks depend on the same Salesforce page remaining in exactly the expected state. Once one test leaves the application in an unexpected state, every following test can become a secondary failure.

Cypress recommends that each test be able to pass independently.

A better pattern is:

```js
beforeEach(() => {
  cy.loginIntoSF('underwriter')
  cy.search_for_id_and_select_case_form('276345725')
})
```

Ideally, make `cy.loginIntoSF()` use `cy.session()` so the authentication itself is cached and you do not pay the full login cost before every test.

Then each test starts from the page/state that it actually requires.

Another reasonable option here is to put all 30 field assertions into **one test** if they are genuinely one logical operation on one already-loaded form. Splitting one stateful page inspection into 30 dependent `it()` blocks gives you very little isolation while making cascading failures much more likely.

I would also turn `testIsolation` back on unless you have a specific reason to disable it.

Please mark this answer as accepted if it helped, thank you.
